interface/billing/era_payments.php

```diff
@@ -233,107 +237,149 @@ function OnloadAction()
         margin-left:5px !Important;
     }
     </style>
+    <?php
+    //to determine and set the form to open in the desired state - expanded or centered
+    // any selection the user makes will become the user-specific default for that page
+    
+    $current_filename = 'era_payments';
+         
+    if (getUserSetting($current_filename) > -1) {
+        $user_value = getUserSetting($current_filename);
+    } elseif ($GLOBALS['expand_form']) {
+        $user_value = $GLOBALS['expand_form'];
+    } else {
+        $user_value = 0;
+    }
+    
+    //the linked files that would need to open in the same state, expanded or centered, as the current page
+    // if no linked files comment out next 4 lines
+    $filenames = array("new_payment", "search_payments", "era_payments");
+    foreach ($filenames as $filename) {
+        setUserSetting($filename, $user_value);
+    }
+    //$user_value = getUserSetting('era_payments');
```

Comment:
Lets work on modularizing this code a bit (ie. centralize the duplicated code)

Turn all of above code into a function in library/user.inc that returns either true (for expand) or false (for center):
```
function collectAndOrganizeExpandSetting ($current_filename, $filenames = array())
```
and then call the function here as an example:
```
$current_state = collectAndOrganizeExpandSetting ("era_payments", array("new_payment", "search_payments", "era_payments"));
```

Reply:
Actually, we can even make this more succinct (since $current_filename will always be included in $filenames; and if it is by itself then the array will just have 1 entry) and work with the block at the end if do:
```
function collectAndOrganizeExpandSetting ($filenames = array()) { // this function will be in library/user.inc }
```
```
$filenamesExpandGroup = array("new_payment", "search_payments", "era_payments");
$current_state = collectAndOrganizeExpandSetting($filenamesExpandGroup);
```

And then you can use the $filenamesExpandGroup array in the expand_contract_js.php script.

Reply:
Done